src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java

```diff
@@ -684,9 +686,14 @@ public Response ondertekenInformatieObject(
     ) {
         final EnkelvoudigInformatieObject enkelvoudigInformatieobject = drcClientService.readEnkelvoudigInformatieobject(
                 uuid);
-        assertPolicy(enkelvoudigInformatieobject.getOndertekening() == null &&
-                     policyService.readDocumentRechten(enkelvoudigInformatieobject,
-                             zrcClientService.readZaak(zaakUUID)).ondertekenen());
+        Zaak zaak = zrcClientService.readZaak(zaakUUID);
+        DocumentRechten documentRechten = policyService.readDocumentRechten(
+                enkelvoudigInformatieobject,
+                zaak
+        );
+        Ondertekening ondertekening = enkelvoudigInformatieobject.getOndertekening();
+        boolean hasOndertekening = ondertekening != null && ondertekening.getDatum() != null;
+        assertPolicy(!hasOndertekening && documentRechten.ondertekenen());
```

Comment:
I would suggest to move `policyService.readDocumentRechten(
                enkelvoudigInformatieobject,
                zaak
        );` back into the `assertPolicy` call. The reason is that we want to keep the policy logic in one place as much as possible. Ideally we even move some of this logic to the OPA Rego policies but that is for another day I would say. :-)
        
```suggestion
        assertPolicy(
           !hasOndertekening && policyService.readDocumentRechten(
                enkelvoudigInformatieobject,
                zaak
            ).ondertekenen()
        );
```
        
ps: you may need to reformat the code after applying this suggestion..